spec/abstractops.html

```diff
@@ -860,14 +860,14 @@ <h1>ISO 8601 grammar</h1>
       FractionalPart :
           DecimalDigit DecimalDigit? DecimalDigit? DecimalDigit? DecimalDigit? DecimalDigit? DecimalDigit? DecimalDigit? DecimalDigit?
 
+      Fraction :
+          DecimalSeparator FractionalPart
+
       TimeFractionalPart :
           FractionalPart
 
-      Fraction :
-          DecimalSeparator TimeFractionalPart
-
       TimeFraction :
-          Fraction
+          DecimalSeparator TimeFractionalPart
```

Comment:
My preference is to minimize redundant productions by e.g. eliminating TimeFractionalPart in favor of keeping TimeFraction : Fraction and referencing TimeFraction when necessary, but this does technically work AFAICT.

Reply:
Fair enough, this was easy enough to do.